### PR TITLE
Add rules for F90 files and some minor changes in fortran flags

### LINKED ISSUE #15616: Let MOOSE make system support F90 files
## Reason
Fortran source files often contain preprocessing derivatives like `#if` or `#include` and the file extension is thus `F90` instead of `f90`. Doxygen follows the same convention for processing Fortran source files.

## Design
But currently MOOSE `build.mk` and `app.mk` do not support `F90` files. If we name all Fortran files with `f90`, Doxygen may not process them properly. It thus is desired to extend MOOSE make system to support `F90` extension.

## Impact
Better mixed language code development.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #15616.

### CODE REVIEW

File: framework/build.mk [hunk 1]
```diff
@@ -240,6 +245,21 @@ ifneq (,$(findstring g95,$(mpif77_command)))
 	libmesh_FFLAGS += -r8
 endif
 
+#ifort
+ifneq (,$(findstring ifort,$(mpif90_command)))
+	libmesh_FFLAGS += -nofpp
```
Comment (loganharbour) [inline framework/build.mk:250]: Why no preprocessor?

**Comment by loganharbour** [comment]
Looks like your compiler changes are causing BISON to diff.

File: framework/build.mk (hunk 1 above)
Comment (YaqiWang) [reply framework/build.mk:250]: I copied this one from my app, but I guess it is not needed.

Comment (loganharbour) [reply framework/build.mk:250]: Yeah - I would say remove it unless we have a specific reason.

I think the bigger concern is the gfortran ones which may be causing BISON to fail.

**Comment by YaqiWang** [comment]
That puzzles me. I might take a look. Could be related with either openmp or 8-bytes real number although both are unlikely.

**Comment by loganharbour** [comment]
Great, thanks. 

I'm content with everything else - it's just the specific ifort and gfortran flags that I would prefer to keep out without a justification with them.

**Pull request closed without merging**